### PR TITLE
Deduplicate Transfer Arg Parsing in Client.

### LINKED ISSUE #1578: Avoid duplicating parsing of transfer args in client
Move the parsing of transfer args from lib.rs to `DeployExt::new_transfer()`.

Also consider removing the `DeployExt` trait, as it probably doesn't give much benefit over free functions.

### PR DESCRIPTION
**Pull request by nilz3ro**
# About These Changes
* closes #1578

### CODE REVIEW

**Review by Fraser999 — APPROVED** [review]

**Review by Fraser999 — APPROVED** [review]

File: client/lib/deploy.rs [hunk 1]
```diff
@@ -251,6 +252,11 @@ impl DeployExt for Deploy {
         const TRANSFER_ARG_TARGET: &str = "target";
         const TRANSFER_ARG_ID: &str = "id";
 
+        let amount = U512::from_dec_str(amount)
+            .map_err(|err| Error::FailedToParseUint("amount", UIntParseError::FromDecStr(err)))?;
```
Comment (rafal-ch) [inline client/lib/deploy.rs]: Could it be that we can use the `const TRANSFER_ARG_AMOUNT` here instead of hardcoding the `"amount"` string?

**Review by rafal-ch — APPROVED** [review]
LGTM!

I left a single minor comment.

Additionally, consider checking if the code that creates `session` in `fn new_transfer(...)` is (or should be) covered by tests. None of the tests in `deploy.rs` uses the mentioned function to create a `Deploy`, they use the `with_payment_and_session` and similar constructors instead.

File: client/lib/deploy.rs (hunk 1 above)
Comment (nilz3ro) [reply client/lib/deploy.rs]: Yes that makes sense.

**Pull request merged**